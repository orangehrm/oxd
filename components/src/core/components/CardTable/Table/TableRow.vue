<template>
  <div :class="classes" role="row">
    <slot></slot>
  </div>
</template>

<script lang="ts">
import {defineComponent} from 'vue';

export default defineComponent({
  name: 'oxd-card-tr',

  props: {
    withBorder: {
      type: Boolean,
      default: true,
    },
    clickable: {
      type: Boolean,
      default: true,
    },
  },

  computed: {
    classes(): object {
      return {
        'oxd-table-row': true,
        'oxd-table-row--with-border': this.withBorder,
        'oxd-table-row--clickable': this.clickable,
      };
    },
  },
});
</script>

<style src="./table-row.scss" lang="scss" scoped></style>
