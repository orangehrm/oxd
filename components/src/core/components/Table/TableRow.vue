<template>
  <tr :class="classes">
    <slot></slot>
  </tr>
</template>

<script lang="ts">
import {defineComponent} from 'vue';

export default defineComponent({
  name: 'oxd-tr',

  props: {
    withBorder: {
      type: Boolean,
      default: true,
    },
    clickable: {
      type: Boolean,
      default: true,
    },
  },

  computed: {
    classes(): object {
      return {
        'oxd-table-row': true,
        'oxd-table-row--with-border': this.withBorder,
        'oxd-table-row--clickable': this.clickable,
      };
    },
  },
});
</script>

<style src="./table-row.scss" lang="scss" scoped></style>
