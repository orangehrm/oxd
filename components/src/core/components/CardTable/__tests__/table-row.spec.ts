import {mount} from '@vue/test-utils';
import TableRow from '@orangehrm/oxd/core/components/CardTable/Table/TableRow.vue';

describe('CardTable > TableRow.vue', () => {
  it('renders OXD CardTable > TableRow', () => {
    const wrapper = mount(TableRow, {
      props: {},
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
