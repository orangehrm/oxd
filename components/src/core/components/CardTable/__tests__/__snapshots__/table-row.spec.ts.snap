// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardTable > TableRow.vue renders OXD CardTable > TableRow 1`] = `<div class="oxd-table-row oxd-table-row--with-border oxd-table-row--clickable" role="row"></div>`;
